### ISSUE TITLE
predicate does not get called on 'user.has_perm(...)'

### ISSUE THREAD

**Issue by Paul-Kijtapart** [dfunckt/django-rules#84]
**Set up**
Django Version: 1.11
Django-rules: v1.3.0 - 2017/12/13

settings.py
```python
AUTHENTICATION_BACKENDS = (
    'rules.permissions.ObjectPermissionBackend',
    'django.contrib.auth.backends.ModelBackend',
)
```

**What should happens:**
After initializing rule set with corresponding predicates, Calling 'has_perm' function on a user should trigger corresponding predicate.

**What actually happens:**
 Calling 'has_perm' function on a user does NOT trigger corresponding predicate, and always return True.

```python

@rules.predicate
def can_view_blog(user):
    print('Calling this rule for user ', user)
    return False

# with initial
rules.add_perm('myapp.view_blogs', can_view_blog)

# this will not trigger can_view_blog, thus returning True instead of False
current_user.has_perm('myapp.view_blogs') 

# However, calling 'has_perm' on rules will find matching permission name and trigger corresponding predicate as expected.
rules.has_perm('myapp.view_blogs')

```

**What I tried:**
I tried both functional and class-based views. However, i cannot seem to trigger the linked predicate on user.has_perm(...)

```python

# import
from rules.contrib.views import permission_required

@rules.predicate
def can_view_blog(user):
    print('Calling this rule for user ', user)
    return False

# link permission with predicate
rules.add_perm('myapp.view_blogs', can_view_blog)

@permission_required('myapp.view_blogs')
def home(request):
    return render(
        request,
        'ems/index.html',
        {
            'title': _('Welcome'),
            'year': timezone.now().year,
        }
    )
```

From the above code,
`rules.has_perm('myapp.view_blogs')` would trigger correct predicate, but `request.user.has_perm('myapp.view_blogs')` will always return True.

**What I think cause this issue:**
1. I thought that this could be a permission name conflict as django-rules use the same name convention as django.contrib.auth.
- Actually this should not be the case as 'rules.permissions.ObjectPermissionBackend' is ordered before contrib.auth, so it will be searched first. Thus, the predicate should gets called in this case.

2. I did not link user to permission system properly, since user.has_perm(any_random_name) will return True.

Any help is appreciated.

**Comment by Paul-Kijtapart** [comment]
**Solution**
user.has_perm does not trigger corresponding predicate, because i am using the user is a superuser. As a result, which always explains why it has_perm always returns True.

**Comment by NotJustPizza** [comment]
It should be documented, I wasted a few hours debugging and searching for info :/

**Comment by dfunckt** [comment]
This is Django’s semantics and have nothing to do with rules. However, PRs are welcome.
